tests/release_tests/test_launch/test_kubernetes.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+import yaml
+from utils import (
+    cleanup_deployment,
+    run_cmd,
+    run_cmd_async,
+    setup_cleanup_on_exit,
+    update_dict,
+    wait_for_image_job_completion,
+)
+from wandb.sdk.launch.launch_add import launch_add
+
+NAMESPACE = "wandb-release-testing"
+ENTITY = "launch-release-testing"
+PROJECT = "release-testing"
+QUEUE = "kubernetes-queue"
+JOB_NAME = "sample_job:v0"  # simple job that counts to 50
+
+LAUNCH_JOB_CONFIG = {
+    "resource_args": {"kubernetes": {"namespace": NAMESPACE}},
+}
+
+AWS_CREDENTIALS_FORMAT = """[default]
+aws_access_key_id = {aws_id}
+aws_secret_access_key = {aws_secret}
+aws_session_token = {aws_token}
+"""
+
+
+@pytest.mark.timeout(180)
+def test_kubernetes_agent_on_local_process():
+    try:
+        # Start launch agent
+        agent_process = run_cmd_async(
+            f"wandb launch-agent -q {QUEUE} -e {ENTITY} -c tests/release_tests/test_launch/local-agent-config.yml"
+        )
+
+        # Start run
+        queued_run = launch_add(
+            job=f"{ENTITY}/{PROJECT}/{JOB_NAME}",
+            queue_name=QUEUE,
+            entity=ENTITY,
+            config=LAUNCH_JOB_CONFIG,
+        )
+
+        status, completed_run = wait_for_image_job_completion(
+            NAMESPACE, ENTITY, PROJECT, queued_run
+        )
+
+        summary = completed_run.summary
+        history = completed_run.history(pandas=False)
+
+        assert (
+            status == "Succeeded"
+        ), "Kubernetes job didn't succeed. Check Kubernetes pods and Docker container output."
+        assert summary["time_elapsed"]
+        assert summary["avg"]
+        assert len(history) == 50
+        assert history[-1]["steps"] == 49
+    finally:
+        agent_process.kill()
+        run_cmd("rm -r artifacts")
+
+
+def _create_config_files():
+    """Create a launch-config.yml and launch-agent.yml."""
+    launch_config = yaml.load_all(
+        open("wandb/sdk/launch/deploys/kubernetes/launch-config.yaml")
```

Comment:
I think the jist is that instead of having `launch-xxxxx-patch.yaml` and `local-agent-config.yml` you should provide a `values.yaml` file for the helm chart and then install the agent with the latest version.

Reply:
This is definitely going to be a follow-up item. For now I'm going with this patching method since I had issues getting helm charts to work with AWS secrets, though that was a while ago so it might be better now.